test/formula.jl

```diff
@@ -365,4 +368,38 @@ module TestFormula
     mm = ModelMatrix(mf)
     @test mm.m[:, 2] == d[complete_cases(d), :x1m]
 
+## Promote non-redundant categorical terms to full rank
+
+d = DataFrame(x = rep([:a, :b], times = 4),
+              y = rep([:c, :d], times = 2, each = 2),
+              z = rep([:e, :f], each = 4))
+[pool!(d, name) for name in names(d)]
+cs = [name => SumContrasts for name in names(d)]
+d[:n] = 1.:8
+
+
+## No intercept
+mf = ModelFrame(n ~ 0 + x, d[1:2,:], contrasts=cs)
```

Comment:
This test is for promoting one variable to full rank when the intercept is missing. It could be re-written using a separate dataframe for each test if that would be clearer...